src/secret_sharing/replicated/malicious/additive_share.rs

```diff
@@ -86,6 +86,21 @@ impl<V: ArithmeticShare> AdditiveShare<V> {
         x: SemiHonestAdditiveShare::ZERO,
         rx: SemiHonestAdditiveShare::ZERO,
     };
+
+    /// Returns share of a scalar value.
+    pub fn share_known_value(
+        helper_role: Role,
+        value: V,
+        r_share: &SemiHonestAdditiveShare<V>,
+    ) -> Self {
+        let x = SemiHonestAdditiveShare::share_known_value(helper_role, value);
+        let rx = match helper_role {
+            Role::H1 => SemiHonestAdditiveShare::new(r_share.left() * value, V::ZERO),
+            Role::H2 => SemiHonestAdditiveShare::share_known_value(helper_role, value),
+            Role::H3 => SemiHonestAdditiveShare::new(V::ZERO, r_share.right() * value),
+        };
```

Comment:
I'm 99% sure this is incorrect.

This is not actually a secret-sharing of r * x. Let's consider the case of x = 1. This will produce a 3-way sharing of the following values:

`H1's r_share.left(), 0, 0`

That is NOT a 3-way secret-sharing of r * 1. That's just one of the 3 random components of r. 

I think what you want is just:

```
let rx = r_share * value;
```

Reply:
r_share is a Replicated<F> so, it would be of form (left, right)
for x = 1, shouldnt the shares be 
H1: ((1, 0), (left * 1, 0))
H2: ((0, 0), (0, 0))
H3: ((0, 1), (0, right * 1))

Here H1's left would be same as H3's right?

Reply:
You are right, fixing it